framework/db/ActiveQuery.php

```diff
@@ -382,8 +396,42 @@ protected function queryScalar($selectExpression, $db)
      */
     public function joinWith($with, $eagerLoading = true, $joinType = 'LEFT JOIN')
     {
-        $this->joinWith[] = [(array) $with, $eagerLoading, $joinType];
+        $relations = [];
+        foreach((array) $with as $name => $callback) {
```

Comment:
no? same as if I would do `foreach($this->getMeSomeArray() as $k => $v)`. This would call the method once.